handshake/crypto_setup_server_test.go

```diff
@@ -100,7 +101,11 @@ type mockStream struct {
 }
 
 func (s *mockStream) Read(p []byte) (int, error) {
-	return s.dataToRead.Read(p)
+	n, _ := s.dataToRead.Read(p)
+	if n == 0 { // block if there's no data
+		time.Sleep(time.Hour)
```

Comment:
Any other suggestion? I don't want to get an `io.EOF` here. This immediately kills the `HandleCryptoStream` in the crypto setups.